### DISCUSSION TITLE
Help with Go debugger

### DISCUSSION THREAD

**Discussion by flo8** [helix-editor/helix#9720] [Troubleshooting]
**Summary**
I have a Go project that compiles just fine using go build. When opening it with Helix and launching the debugger, an error happens (see log below). The autocompletion works well. Only debugging doesn't seem to work even if dlv is detected.

**Reproduction steps:**

Go is properly installed on the system (1.22) with `hx --health go` giving the following output:
```
Configured language servers:
  ✓ gopls: /Users/flo/go/bin/gopls
Configured debug adapter: dlv
Binary for debug adapter: /Users/flo/go/bin/dlv
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```

Open any simple Go project/file that compiles. Then press **< space > gl** and then choose Source and main.go file
(this was advised here https://github.com/helix-editor/helix/discussions/6903)

I **didn’t** modify the `languages.toml` and using the default Helix config.

**I expected this to happen:**
Start the debugging session until my first breakpoint.

**Instead, this happened:**
I get the following error in the statusbar:
`Async job failed: Some(Object {"error": Object {"format": String("Failed to launch: Build error: Check the debug console for details."), "id": Number(3000), "showUser": Bool(false)}})`

Helix log:
```
2024-02-25T10:06:27.371 helix_dap::transport [ERROR] <- DAP error Some("Failed to launch") (Some(Object {"error": Object {"format": String("Failed to launch: Build error: Check the debug console for details."), "id": Number(3000), "showUser": Bool(false)}})) for command #1 launch
2024-02-25T10:06:27.372 helix_view::editor [ERROR] editor error: Async job failed: Some(Object {"error": Object {"format": String("Failed to launch: Build error: Check the debug console for details."), "id": Number(3000), "showUser": Bool(false)}})
2024-02-25T10:06:31.184 helix_dap::transport [ERROR] err: <- StreamClosed
2024-02-25T10:06:31.185 helix_lsp::transport [ERROR] gopls err: <- StreamClosed
```

**Platform**
macOS

**Terminal Emulator**
iTerm2

**Installation Method**
Brew

**Helix Version**
23.10 (f6021dd0)

Am I missing something in how to use the debugger with Helix? The very same project runs the debugger just fine in VSCode.

**Note:** I preferred to put this in discussion rather than issues as I am not sure this is a bug/issue yet.

**Comment by gnp-x** [comment]
Worked fine for me in macOS in iTerm2. 
- I did `<space> gn`
- Toggled a breakpoint in my main.go file
- Launched debug target
- Selected source, typed main.go, because that's where I toggled my breakpoint. 

Then it begins to debug on my breakpoint, where I can step in, out, next, and other things with the debug menu. Maybe one of those steps you skipped?

Keep in mind, debugging in helix is experimental, so there may be issues with using it.
